windows/src/global/vc/xstring.cpp

```diff
@@ -47,7 +47,7 @@ PWSTR incxstr(PWSTR p)
 		case CODE_INDEX:		return p+2;
 		case CODE_USE:			return p+1;
 		case CODE_DEADKEY:		return p+1;
-		case CODE_EXTENDED:		p += 2; while(*p != UC_SENTINEL_EXTENDEDEND) p++; return p+1;
+		case CODE_EXTENDED:		p += 2; while(*p && *p != UC_SENTINEL_EXTENDEDEND) p++; return p+1;
```

Comment:
This fixes a buffer overrun on malformed xstrings. I need to make the same change in Keyman Engine and verify Keyman Core also.